✨ Feat : 주문생성/삭제 동시성 제어

yejin0718 opened the pull request:
<!-- 
체크하려면 괄호 안에 "x"를 입력하세요. 
각 규칙은 Convention 문서에 있습니다.
PR 제목에 쓰는 prefix는 다음과 같습니다.

✨ Feat : 새로운 기능
🔨️ Refactor : 코드 리팩토링
🐎 Perf : 성능을 향상
🐛 Fix : 버그를 고칠 때
🧪 Test : 테스트 코드
🚜 Rename : 파일 이름 변경 혹은 구조를 변경
🚀 Release : 배포 / 개발 작업과 관련된 모든 것
🔥 Remove : 코드 또는 파일 제거
📚 Docs : 문서
📝 Chore : 사소한 코드 또는 언어를 변경 기타 변경사항 (빌드 스크립트 수정, 패키지 매니징 설정 등)

-->

## PR 타입
<!-- 어떤 유형의 PR인지 체크해주세요. -->

<!-- 체크하려면 괄호 안에 "x"를 입력하세요. -->
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Performance Improvement
- [ ] Bugfix
- [ ] Test Code
- [ ] Code style update (formatting, local variables)
- [ ] Other... Please describe:



## 개요 
- 주문생성/삭제 동시성 제어 추가


## 작업 및 변경 사항 
- RedissonLockFacade작성
- 테스트 코드 작성
- controller 수정


### 테스트 결과 
- 내용을 적어주세요.


### 공유 하고 싶은 내용 및 참고 자료
- 내용을 적어주세요.


close #62

yejin0718 committed:
Feat: redis 의존성 추가

yejin0718 committed:
feat: 주문생성 RedissonLockRacade 작성

yejin0718 committed:
feat: 주문생성 테스트코드 작성

yejin0718 committed:
feat: 주문삭제 redissonLockFacade 작성

yejin0718 committed:
feat: 주문삭제 테스트 코드 작성

yejin0718 committed:
refactor: orderController 수정

hyunmuam commented on src/main/java/com/hanbang/e/order/facade/OrderRedissonLockFacade.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    public void insertOrder(Long key, Long memberId, Long productId, OrderReq orderReq){
+        String inserKey = String.valueOf(1+key);
```
P1: `insertKey`를 `inserKey`로 오타난 것 같습니다. 확인부탁드립니다.

yejin0718 replied:
네 수정하겠습니다!

hyunmuam commented on src/main/java/com/hanbang/e/order/controller/OrderController.java:
```diff
@@ -49,8 +55,9 @@ public ResponseEntity<ResponseDto<List<OrderResp>>> getMyOrderList(HttpServletRe
-		orderService.deleteOrder(memberId, orderId);
+		orderRedissonLockFacade.deleteOrder(key, memberId, orderId);
 		return new ResponseEntity<>(new ResponseDto("success", "주문 삭제 성공", null), HttpStatus.OK);
 	}
```
P2: 파사드 패턴을 쓰신 것 같은데, 파사드 패턴을 Controller계층에서 직접 접근하는것이 맞는지 한번 알아봐야 될 것 같습니다.

yejin0718 replied:
넵 확인해보고 수정하겠습니다

hyunmuam reviewed (changes requested):
P1: 코멘트 확인 부탁드립니다.

hyunmuam committed:
[feat] Redisson 분산락 annotation 기반으로 사용하기 기능 구현

@DistributeLock을 동시성문제가 일어날 수 있는 메소드에 annotation으로 붙이면 동시성 제어를 해주는 기능 추가

Related to: #62, #64

hyunmuam committed:
[test] Redisson 분산락 테스트코드 추가

분산락이 잘 작동되는지 확인, 테스트 결과 이상없음

Resolves: #64
Related to: #62

yejin0718 committed:
rename: 오타 수정

hyunmuam committed:
[rename] distributeLock관련 클래스 패키지 변경 / 테스트코드 위치 변경

hyunmuam committed:
✨ Feat : Redisson 분산락 annotation 기반으로 사용하기 기능 추가

yejin0718 committed:
refactor: 주문생성 redis 어노테이션으로 수정

hyunmuam committed:
[test] 동시성 제어 테스트코드 추가

- 회원 가입 동시성 제어 테스트코드 추가
- 상품 주문, 주문취소 기능의 동시성 제어 테스트코드 추가

Resolves: #62
Ref: #64

hyunmuam committed:
Merge branch 'develop' into feature/#62

hyunmuam committed:
[add] 레디스 연결정보 수정

yanJuicy committed:
[ci] CI 과정에서 docker 실행

yanJuicy committed:
[refactor] redis host 변경

yanJuicy committed:
[refactor] resis host 변경

yanJuicy committed:
[refactor]: ci 실행 시 docker-compose.redis 실행하도록 변경

hyunmuam committed:
[fix] gradle 빌드 실패로 인한 테스트 코드 수정
